src/containers/SearchProvider/SearchProvider.js

```diff
@@ -31,14 +32,27 @@ class SearchProvider extends Component {
   };
 
   runQuery = () => {
-    const { client } = this.props;
+    const { client }                                            = this.props;
+    const { queryString, searchPhrase, searchTags, categories } = this.state;
+
+    if (searchTags.length === 0 && searchPhrase !== '') {
+      this.setState({ queryString: searchPhrase });
+    } else if (searchTags.length > 0 && searchPhrase === '') {
+      const searchTagsToString = searchTags.join(" ");
+
+      this.setState({ queryString: searchTagsToString });
+    } else if (searchTags.length > 0 && searchPhrase !== '') {
+      const searchTagsToString = searchTags.join(" ");
+
+      this.setState({ queryString: `${searchTagsToString} ${searchPhrase}` });
+    }
 
     client
       .query({
         query    : SEARCH_QUERY,
         variables: {
-          searchPhrase: this.state.searchPhrase,
-          categories  : this.state.categories,
+          searchPhrase: queryString,
+          categories  : categories,
```

Comment:
If think this could be as simple as;

```js
  runQuery = () => {
    const { client } = this.props;
    const { searchPhrase, searchTags, categories } = this.state;

    client
      .query({
        query    : SEARCH_QUERY,
        variables: {
          searchPhrase: searchTags.concat(searchPhrase).join(' '),
          categories,
```

Or if you would like to use this "composed" search query, this could also be determined in the `handleSubmit` function.

```js
  handleSearchSubmit = (event, searchPhrase, searchTags) => {
    const searchQuery = searchTags.concat(searchPhrase).join(' '),

    this.setState({ searchPhrase, searchTags, searchQuery });
    this.runQuery();
  };

  runQuery = () => {
    const { client } = this.props;
    const { searchQuery, categories } = this.state;

    client
      .query({
        query    : SEARCH_QUERY,
        variables: {
          searchPhrase: seachQuery,
          categories, 
```